### PR TITLE
Final Tracker Geometry for tracker TDR + Cleaning old Phase1

### PR DESCRIPTION
**Pull request by boudoul**
Greetings 
This pull request aim for 
- Describing the final Tracker for the TDR (in D11) described here : 
http://ghugo.web.cern.ch/ghugo/layouts/2016/OT613_200_IT4025/

- Fix the important thing which was a (non desired and problematic) dependencies of the phase2 Geometry with Phase1 Geometry (for the inner pixel part ) 
--> the Inner Tracker is now defined as P2PXBBarrel and no more as P1PXBBarrel which could cause severe bugs for any changes in 2017 geometry 

- The usage of  P1PXBBarrel eleswhere in cmssw has been tracked down and fixed accordingly 

- Note that the D11 is no more running the track trigger sequence, as it should be fix ot deal with the new geometries (new rings) to be fix in a subsequent PR 

- I have added a protection in the phase2 DQM in case of no valid collections are found 

- PLease note that this PR (in particular D11 sceanrio ) should be tested with this PR :+1: https://github.com/cms-data/SLHCUpgradeSimulations-Geometry/pull/6

### CODE REVIEW

**Comment by boudoul** [comment]
I have tested locally but i'm planning to provide with more stats some relevant
tracking plots in a couple of days , but it doesn't prevent the review to start
- thanks !

**Comment by boudoul** [comment]
adding @delaere , @atricomi

**Comment by kpedro88** [comment]
@boudoul can you summarize the "real" changes in the T5/D11 geometry (as opposed to technical reorganizations)?

File: Geometry/TrackerCommonData/data/PhaseII/TiltedTracker4025/pixfwd.xml [hunk 1]
```diff
@@ -47,34 +47,10 @@ note: see OT365_200_IT4025_2016_12_06.cfg for full config files
...
-   <ZSection z="[RootMidZ2]"   rMin="[cms:TrackBeamR1]" rMax="[RootRadius]"/>
-   <ZSection z="[RootMidZ3]"   rMin="[cms:TrackBeamR1]" rMax="[RootRadius]"/>
-   <ZSection z="[RootEndZ]"    rMin="[cms:TrackBeamR2]" rMax="[RootRadius]"/>
-   </Polycone>-->
  <Polycone name="Forward" startPhi="0*deg" deltaPhi="360*deg" >    <!-- TEMPORARY!! Should be defined automatically -->
    <ZSection z="[RootStartZ]" rMin="2.8*cm" rMax="[RootRadius1]"/>
-   <ZSection z="99.00*cm" rMin="2.8*cm" rMax="[RootRadius1]"/>
-   <ZSection z="99.00*cm" rMin="2.8*cm" rMax="116.0*cm"/>
+   <ZSection z="95.90*cm" rMin="2.8*cm" rMax="[RootRadius1]"/>
+   <ZSection z="95.90*cm" rMin="2.8*cm" rMax="116.0*cm"/>
```
Comment (ianna) [inline Geometry/TrackerCommonData/data/PhaseII/TiltedTracker4025/pixfwd.xml]: @boudoul - Have you checked that this move in Z by 3.1 cm does not introduce an overlap? Or shall I do it? Thanks.

Comment (boudoul) [inline Geometry/TrackerCommonData/data/PhaseII/TiltedTracker4025/pixfwd.xml]: @ianna ,   @ghugo should have checked that, maybe she can confirm

**Comment by boudoul** [comment]
@kpedro88 Sure, I was exacly preapring this, comments in github will be there in a few minutes

**Comment by iahmad-khan** [comment]
please test with cms-sw/cmsdist#2942

**Comment by ebrondol** [comment]
Hi @boudoul and @atricomi, the cluster shape filter is not been updated. Am I correct? I suppose this is the best way to have it now, then if we see that we need to update it, it can be done in a future PR.

File: Alignment/CommonAlignment/src/AlignableObjectId.cc [hunk 2]
```diff
@@ -308,8 +308,6 @@ align::StructureType AlignableObjectId::stringToId(const char *name) const
 //______________________________________________________________________________
 AlignableObjectId::Geometry AlignableObjectId
 ::trackerGeometry(const TrackerGeometry* geometry) {
-  if (!geometry) return Geometry::General;
```
Comment (ghellwig) [inline Alignment/CommonAlignment/src/AlignableObjectId.cc]: @boudoul here you re-introduce I bug I fixed in #18097

**Review by ghellwig — CHANGES_REQUESTED** [review]
@boudoul Except for the re-introduced bug, I am fine with the changes.

File: Alignment/CommonAlignment/src/AlignableObjectId.cc (hunk 2 above)
Comment (boudoul) [inline Alignment/CommonAlignment/src/AlignableObjectId.cc]: whoops will fix in the next commit - thanks for noticing !

**Comment by boudoul** [comment]
Description : of the changes in T5 : 
1) Outer Tracker PS modules:
PS-p active length: 32 x 1467 = 46944 mu. Used to be 32 x 1446.

2) Outer Tracker Barrel PS layers : 
+ 1 ring in TBPS Layer 1.
Now, all TBPS layers have 12 rings.
Minor adjustments in coordinates in TBPS.

3) Outer Tracker EndCaps: 
Disk 1 has a much lower Z.
PS instead of 2S in Ring 10 (disks 1 to 5).
- 1 ring in TEDD (disks 3 to 5).
Now, in TEDD, disks 3 to 5 have 12 rings each. 

4) Minor update in pixel IT MB 

I let @ghugo check if my list is complete

**Comment by boudoul** [comment]
@ebrondol , you are correct and +1 to your comments, thanks !
Le 7 avr. 2017 à 11:39, ebrondol <notifications@github.com<mailto:notifications@github.com>> a écrit :


Hi @boudoul<https://github.com/boudoul> and @atricomi<https://github.com/atricomi>, the cluster shape filter is not been updated. Am I correct? I suppose this is the best way to have it now, then if we see that we need to update it, it can be done in a future PR.

**Comment by boudoul** [comment]
Informing @lmoreels for the changes i'm introducing here in the DQM phase2 since she's the original developer

**Comment by kpedro88** [comment]
@boudoul thanks

@ebrondol can you or someone else do a 200PU igprof run to see if the possibly non-ideal cluster shape filter impacts CPU performance?

File: Geometry/TrackerCommonData/data/PhaseII/TiltedTracker4025/pixfwd.xml (hunk 1 above)
Comment (ghugo83) [inline Geometry/TrackerCommonData/data/PhaseII/TiltedTracker4025/pixfwd.xml]: Hi, yes i modifed to solve extrusions (first disk in TEDD moved inwards).

**Comment by ghellwig** [comment]
@ianna there should be a corresponding GT with the updated geometry payloads, right?
Can you take care of this?

**Comment by kpedro88** [comment]
@ghellwig for phase2 we're just using XML geometries right now

If any of the changes affect phase1 then the 2017 tag should be updated

**Comment by ianna** [comment]
@kpedro88 - yes, but @ghellwig is right - there is a GT and it should be updated (IMHO). I'll take care of it.

File: Geometry/TrackerCommonData/data/PhaseII/TiltedTracker4025/pixfwd.xml (hunk 1 above)
Comment (ghugo83) [inline Geometry/TrackerCommonData/data/PhaseII/TiltedTracker4025/pixfwd.xml]: On a more general note, once I finished these TDR XMLs, I did a first-glance validation in CMSSW to solve all extrusions / overlaps with the following tools : 
- Fireworks;
- Geant4 overlap checking tool.
Any cross-check is welcome though :)

**Comment by ghugo83** [comment]
@boudoul Yes I confirm these are the changes in the geometry that correspond to these XMLs.

As regards the point on the ring in TEDD, in disks 3 to 5 : we have one ring less (12 rings instead of 13).
TEDD disks 1 and 2 still have 15 rings each.
Geometric coverage was improved in TBPS and TEDD.

**Comment by ghugo83** [comment]
I did a MB comparison between CMSSW Sim and tkLayout for this layout, will post it on Monday ;)

**Comment by ghellwig** [comment]
thanks @ianna for taking care of it!

**Comment by ghellwig** [comment]
+1

**Comment by ianna** [comment]
+1

**Comment by ghugo83** [comment]
Material Budget comparison between CMSSW Sim and tkLayout is enclosed (for D11).
This is within Full Tracker volume, Beam Pipe excluded. 10000 tracks.
materials:Air is not taken into account here.

ratio

**Comment by kpedro88** [comment]
+1
2023 workflows show differences in the comparisons, primarily for tracker/tracking and related quantities, due to geometry changes. Other workflow results are unchanged.

**Comment by civanch** [comment]
+1

**Comment by dmitrijus** [comment]
+1

**Comment by ianna** [comment]
@ghellwig - the following tags are available in DB for 2023 D11 tracker:
```
XMLFILE_Geometry_91YV5_Extended2023D11_mc
TKRECO_Geometry_91YV5
TKExtra_Geometry_91YV5
TKParameters_Geometry_91YV5
```
@kpedro88 - BTW, I noticed that HCal RECO geometry size is much bigger then it used to be. Is it really so detailed? Here is a comparison with other payloads:
```
-rw-r--r--. 1 yana zh    61440 Apr 11 16:07 CASTORRECO_Geometry.db
-rw-r--r--. 1 yana zh    57344 Apr 11 16:07 CSCRECODIGI_Geometry.db
-rw-r--r--. 1 yana zh   151552 Apr 11 16:07 CSCRECO_Geometry.db
-rw-r--r--. 1 yana zh   163840 Apr 11 16:07 CTRECO_Geometry.db
-rw-r--r--. 1 yana zh   688128 Apr 11 16:07 DTRECO_Geometry.db
-rw-r--r--. 1 yana zh  2015232 Apr 11 16:07 EBRECO_Geometry.db
-rw-r--r--. 1 yana zh   425984 Apr 11 16:07 GEMRECO_Geometry.db
-rw-r--r--. 1 yana zh  2289664 Apr 11 16:07 GeometryFileExtended2023D11.db
-rw-r--r--. 1 yana zh    61440 Apr 11 16:07 HCALParameters_Geometry.db
-rw-r--r--. 1 yana zh 27328512 Apr 11 16:07 HCALRECO_Geometry.db
-rw-r--r--. 1 yana zh   598016 Apr 11 16:07 RPCRECO_Geometry.db
-rw-r--r--. 1 yana zh  2535424 Apr 11 16:07 TKExtra_Geometry.db
-rw-r--r--. 1 yana zh    57344 Apr 11 16:07 TKParameters_Geometry.db
-rw-r--r--. 1 yana zh 12500992 Apr 11 16:07 TKRECO_Geometry.db
-rw-r--r--. 1 yana zh    57344 Apr 11 16:07 ZDCRECO_Geometry.db
```

**Comment by kpedro88** [comment]
@ianna better to ask @bsunanda about HCAL RECO geometry tag size

**Comment by ianna** [comment]
@kpedro88 and @bsunanda - the tag size is proportional to the number of DetIds in Hcal reco geometry. My question is how many DetIds are there in 2023 D11 Hcal - I presume HB only?

**Comment by kpedro88** [comment]
@ianna ah, I think the Hcal DetIds are also used for HE = BH, even though the digi/reco is done in HGCal code

**Comment by ianna** [comment]
@kpedro88 and @bsunanda - How much has HCal reco geometry changed since 810? The HCal reco payload size in 810 is 376832 (compared to 27328512 now).

**Comment by kpedro88** [comment]
@ianna do you mean specifically for Phase2? I don't really remember what was in 810 for Phase2.

**Comment by ianna** [comment]
@kpedro88 - D1 - D6

**Comment by kpedro88** [comment]
hm, I would think C2 hasn't changed since then.

**Comment by ianna** [comment]
@ghellwig - Please use a candidate GT: 91X_upgrade2023_realistic_Candidate_2017_04_11_15_56_35, D11 scenario is set a default geometry (e.g. label Extended).
Alignments need to be updated:
```
Exception Message:
Size mismatch between geometry (size=44240) and alignments (size=19876)
```

**Comment by boudoul** [comment]
Hi @ghellwig , @ianna , I'm not sure what we are trying to achive here  - As pointed by @kpedro88 , we are not using the geometry GT but xmls for any phae2 scenario . There's no alignement to be applied (this is exactly why there is this line  : https://github.com/cms-sw/cmssw/blob/master/Configuration/Geometry/python/GeometryExtended2023D11Reco_cff.py#L14 ) -

**Comment by boudoul** [comment]
and for a previous comment from @kpedro88 : 
" If any of the changes affect phase1 then the 2017 tag should be updated "
the changes proposed here are NOT affecting 2017

**Comment by ianna** [comment]
@boudoul - there is an upgrade2023 GT queue and even if it's not used, it should not diverge too much from a transient geometry.

**Comment by ianna** [comment]
@kpedro88 and @bsunanda - I've moved the question about HCal reco geometry payload to https://github.com/cms-sw/cmssw/issues/18326

**Comment by perrotta** [comment]
+1
Mostly geometry: reco updates are kind of simple
Changes are only for phase2 wfs, as expected

**Comment by boudoul** [comment]
Thanks to all reviewers ! Can PdmV take a look @fabozzi , @GurpreetSinghChahal, ? Thanks
